core/drivers/scmi-msg/platform_weaks.S

```diff
@@ -0,0 +1,56 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2019, Linaro Limited
+ */
+#include <arm32_macros.S>
+#include <asm.S>
+#include <drivers/scmi.h>
+#include <kernel/unwind.h>
+
+	/*
+	 * Weak functions for SCMI message driver API the platform does
+	 * not implement.
+	 */
+
+#ifdef CFG_SCMI_MSG_CLOCK
+	.weak plat_scmi_clock_count
+	.weak plat_scmi_clock_get_name
+	.weak plat_scmi_clock_rates_array
+	.weak plat_scmi_clock_rates_by_step
+	.weak plat_scmi_clock_get_current_rate
+	.weak plat_scmi_clock_set_current_rate
+	.weak plat_scmi_clock_get_state
+	.weak plat_scmi_clock_set_state
+#endif
+
+/* Unique weak function for unsupported handlers returning SCMI_NOT_SUPPORTED */
+FUNC scmi_weak_not_supported , :
+
+#ifdef CFG_SCMI_MSG_CLOCK
+plat_scmi_clock_set_current_rate:
```

Comment:
This is implemented in assembly because we don't want to use
` __attribute__ ((weak, alias ("scmi_weak_not_supported")));` in C?

Reply:
I used assembly to gather all weak functions into 2 functions to optimize a bit the memory footprint.

Reply:
The attribute stuff I wrote is the C equivalent which I expect will do exactly the same thing, but with the advantage of avoiding duplicated code for ARM32/64.

Reply:
ok, i'll look at it. With your suggestion, I think the aliased functions must shared the very same prototype. The proposal here factorizes functions that have different prototypes: argument list changes but they only share the same return value.